docs/source/branching-strategy.rst

```diff
@@ -0,0 +1,41 @@
+The branching strategy
+----------------------
+
+Here are some guidelines on our usage of branches.
+
+**Create a new N.x branch whenever a new mainnet deployment is made.**
+Such a branch is called a *deployment branch*. This clearly separates different
+mainnet deployments and makes it easier to test agent and the frontend for a
+particular deployment. The new branch number N is always 1 greater than the
+number of the most recently created deployment branch.
+
+**The most recent deployment branch follows the main branch until someone
+decides to diverge from main.**
+This can be done as easily as::
+
+    git push origin main:N.x
+
+Note that this must result in a fast-forward of the ``N.x`` branch.
+
+**After a deployment branch diverges from main, further changes to the deployment branch
+require PRs to be made.**
+This is to ensure proper consideration and code review of the changes. In most cases it is
+expected to have commits from ``main`` cherry-picked to ``N.x``.
```

Comment:
> This might be a little bit confusing as in these cases we will end up with `N-1.x` & `N.x` branches where the PR enforcing is to be done on the `N-1.x` branch and not on the `N.x` right? Or maybe i got it wrong by thinking that "diverging from main" means immediate creation of a new branch `N.x`?

Creation of a new `N.x` branch is done only immediately after a new deployment is done. From that moment on, the branch `N-1.x` is not expected to be updated -- the maintenance now moves to the newly-created `N.x` branch.
"Diverging from main" could perhaps better be named "stopping following main"; it does not mean that a new branch is created (that was done after the deployment), rather it means we no longer sync the deployment branch with `main`, but rather pick individual changes via separate PRs.

Reply:
Thanks for explaining :+1: